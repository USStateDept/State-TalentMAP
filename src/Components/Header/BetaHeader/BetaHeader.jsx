// Once About page is ready, uncomment the commented code
import React from 'react';
// import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';

const BetaHeader = () => (
  <div className="usa-banner tm-beta-header">
    <div className="usa-grid usa-banner-inner padded-main-content">
      <FontAwesome name="gears" />
      Welcome to TalentMAP!  Use this site to research positions and continue to bid using FSBid.
       New features are coming soon.
      {/* <Link to="/about">Read more about how to use this site and what&#39;s to come.</Link> */}
    </div>
  </div>
);

export default BetaHeader;
