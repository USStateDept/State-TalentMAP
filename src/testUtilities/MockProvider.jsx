import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { FlagsProvider } from 'flag';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import getFlags from '../flags';

const flags = () => getFlags();
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

export const MockTestProvider = ({ flagProps, providerProps, memoryProps, children }) => (
  <FlagsProvider flags={flags()} {...flagProps}>
    <Provider store={mockStore({})} {...providerProps}>
      <MemoryRouter {...memoryProps}>
        {children}
      </MemoryRouter>
    </Provider>
  </FlagsProvider>
);

MockTestProvider.propTypes = {
  flagProps: PropTypes.shape({}),
  providerProps: PropTypes.shape({}),
  memoryProps: PropTypes.shape({}),
  children: PropTypes.node.isRequired,
};

MockTestProvider.defaultProps = {
  flagProps: {},
  providerProps: {},
  memoryProps: {},
};

export default MockTestProvider;
