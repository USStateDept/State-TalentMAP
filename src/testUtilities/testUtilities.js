import sinon from 'sinon';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import MockAdapter from 'axios-mock-adapter';
import api from '../api';

// Tests all functions within a mapDispatchToProps function.
// We take two arguments, one is the entire mapDispatchToProps function, which
// can be exported from the Container/Component. The second is a config
// object with any parameters we want to pass to the function with the same name.
// For example, to pass the argument "test" to the getData() function, and "test"
// and {1} to submitData() function, we could create an object such as
// { getData: ["test"], submitData: ["test", 1] }
// Then we spy dispatch and call each property of the mapDispatchToProps
// object. Calling each property should trigger the call count on the spy,
// which we test. We reset call count between tests since we're using the
// calledOnce method.
export function testDispatchFunctions(mapDispatchToProps, config = {}) {
  it('should call all mapDispatchToProps actions', () => {
    // create spy
    const dispatchSpy = sinon.spy();
    // object with all functions
    const spiedProps = mapDispatchToProps(dispatchSpy);
    // iterate on each key
    Object.keys(spiedProps).forEach((prop) => {
      // set the prop to test
      const propToTest = spiedProps[prop];
      // set any params passed in the config object
      let params = [];
      // if there any arguments for that param, set them
      if (config[prop]) { params = config[prop]; }
      // call the prop
      propToTest(...params);
      // ensure it was called once
      sinon.assert.calledOnce(dispatchSpy);
      // reset so that called count is 0
      dispatchSpy.reset();
    });
  });
}

export function setupAsyncMocks() {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const mockAdapter = new MockAdapter(api);

  return { mockStore, mockAdapter };
}
