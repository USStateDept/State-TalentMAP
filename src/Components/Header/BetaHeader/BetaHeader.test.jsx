import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import { testDispatchFunctions } from '../../../testUtilities/testUtilities';
import BetaHeader, { mapDispatchToProps } from './BetaHeader';

describe('BetaHeader', () => {
  it('is defined', () => {
    const wrapper = shallow(
      <BetaHeader.WrappedComponent />,
    );
    expect(wrapper).toBeDefined();
  });

  it('calls patchData on this.patchData()', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <BetaHeader.WrappedComponent patchData={spy} />,
    );
    wrapper.instance().patchData();
    sinon.assert.calledOnce(spy);
  });

  it('updates state on this.toggleEditor()', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <BetaHeader.WrappedComponent patchData={spy} />,
    );
    expect(wrapper.instance().state.editorVisible).toBe(false);
    wrapper.instance().toggleEditor();
    expect(wrapper.instance().state.editorVisible).toBe(true);
  });

  it('updates state and calls patchData on this.submit()', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <BetaHeader.WrappedComponent patchData={spy} />,
    );
    wrapper.instance().setState({ editorVisible: true });
    expect(wrapper.instance().state.editorVisible).toBe(true);
    wrapper.instance().submit();
    expect(wrapper.instance().state.editorVisible).toBe(false);
    sinon.assert.calledOnce(spy);
  });

  it('updates state on this.cancel()', () => {
    const spy = sinon.spy();
    const wrapper = shallow(
      <BetaHeader.WrappedComponent patchData={spy} />,
    );
    wrapper.instance().setState({ editorVisible: true });
    expect(wrapper.instance().state.editorVisible).toBe(true);
    wrapper.instance().cancel();
    expect(wrapper.instance().state.editorVisible).toBe(false);
  });

  it('displays one message when there are no line breaks', () => {
    const wrapper = shallow(
      <BetaHeader.WrappedComponent data="test" />,
    );
    expect(wrapper.find('.header-row').length).toBe(1);
  });

  it('displays one message when there are three line breaks and state.headerDropdownVisible is false', () => {
    const data = 'test\n\n\ntest';
    const wrapper = shallow(
      <BetaHeader.WrappedComponent data={data} />,
    );
    expect(wrapper.find('.header-row').length).toBe(1);
  });

  it('displays two messages when there are three line breaks and state.headerDropdownVisible is true', () => {
    const data = 'test\n\n\ntest';
    const wrapper = shallow(
      <BetaHeader.WrappedComponent data={data} />,
    );
    wrapper.instance().setState({ headerDropdownVisible: true });
    wrapper.update();
    expect(wrapper.find('.header-row').length).toBe(2);
  });

  it('displays five messages when there are three line breaks and state.headerDropdownVisible is true', () => {
    const data = 'test\n\n\n  test  test 2 \n\n\ntest3\n\n\n a test\n\n\ntest  3 4 \n\n'; // add just two line breaks at end, which shouldn't trigger new message
    const wrapper = shallow(
      <BetaHeader.WrappedComponent data={data} />,
    );
    wrapper.instance().setState({ headerDropdownVisible: true });
    wrapper.update();
    expect(wrapper.find('.header-row').length).toBe(5);
  });

  it('displays the correct button text when state.headerDropdownVisible is changed', () => {
    const data = 'test\n\n\ntest';
    const wrapper = shallow(
      <BetaHeader.WrappedComponent data={data} />,
    );
    expect(wrapper.find('button').text()).toBe('2Show alerts <FontAwesome />');
    wrapper.instance().setState({ headerDropdownVisible: true });
    wrapper.update();
    expect(wrapper.find('button').text()).toBe('2Hide alerts <FontAwesome />');
  });

  it('displays the correct button text when the button is clicked', () => {
    const data = 'test\n\n\ntest';
    const wrapper = shallow(
      <BetaHeader.WrappedComponent data={data} />,
    );
    expect(wrapper.find('button').text()).toBe('2Show alerts <FontAwesome />');
    wrapper.find('button').simulate('click');
    expect(wrapper.find('button').text()).toBe('2Hide alerts <FontAwesome />');
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <BetaHeader.WrappedComponent />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    const config = {
      fetchData: [],
      patchData: [{}],
    };
    testDispatchFunctions(mapDispatchToProps, config);
  });
});
