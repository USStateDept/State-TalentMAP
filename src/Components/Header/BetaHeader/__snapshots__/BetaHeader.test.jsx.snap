// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetaHeader matches snapshot 1`] = `
<div
  className="usa-banner tm-beta-header"
>
  <div
    className="usa-grid usa-banner-inner padded-main-content"
  >
    <FontAwesome
      name="gears"
    />
    Welcome to TalentMAP!  Use this site to research positions and continue to bid using FSBid. New features are coming soon.
  </div>
</div>
`;
